app/controllers/sessions_controller.rb

```diff
@@ -105,16 +105,13 @@ def do_login
 
   def login_failure(message)
     StatsD.increment "login.failure"
+    flash.now.notice = message
+    render "sessions/new", status: :unauthorized
+  end
 
-    respond_to do |format|
-      format.json do
-        render json: { message: message }, status: :unauthorized
-      end
-      format.html do
-        flash.now.notice = message
-        render template: "sessions/new", status: :unauthorized
-      end
-    end
```

Comment:
splitted this up into `login_failure` and `webauthn_verification_failure`. Originally, the json branch would only apply for webauthn verification failures while the html would apply to other login failures.